api/service/stagedsync/stage_blockhashes.go

```diff
@@ -0,0 +1,600 @@
+package stagedsync
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/harmony-one/harmony/core"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon-lib/kv/mdbx"
+	"github.com/ledgerwatch/log/v3"
+)
+
+type StageBlockHashes struct {
+	configs StageBlockHashesCfg
+}
+
+type StageBlockHashesCfg struct {
+	ctx           context.Context
+	bc            core.BlockChain
+	db            kv.RwDB
+	turbo         bool
+	turboModeCh   chan struct{}
+	bgProcRunning bool
+	isBeacon      bool
+	cachedb       kv.RwDB
+}
+
+func NewStageBlockHashes(cfg StageBlockHashesCfg) *StageBlockHashes {
+	return &StageBlockHashes{
+		configs: cfg,
+	}
+}
+
+func NewStageBlockHashesCfg(ctx context.Context, bc core.BlockChain, db kv.RwDB, isBeacon bool, turbo bool) StageBlockHashesCfg {
+	cachedb, err := initHashesCacheDB(ctx, isBeacon)
+	if err != nil {
+		panic("can't initialize sync caches")
+	}
+	return StageBlockHashesCfg{
+		ctx:      ctx,
+		bc:       bc,
+		db:       db,
+		turbo:    turbo,
+		isBeacon: isBeacon,
+		cachedb:  cachedb,
+	}
+}
+
+func initHashesCacheDB(ctx context.Context, isBeacon bool) (db kv.RwDB, err error) {
+	// create caches db
+	cachedbName := Block_Hashes_Cache_DB
+	if isBeacon {
+		cachedbName = "beacon_" + cachedbName
+	}
+	cachedb := mdbx.NewMDBX(log.New()).Path(cachedbName).MustOpen()
+	tx, errRW := cachedb.BeginRw(ctx)
```

Comment:
what does tx stand for? transaction? It's a bit confusing. Better rename or add comment to clarify.

Reply:
comments added!